single_kernel_mongo/managers/mongodb_operator.py

```diff
@@ -685,18 +777,10 @@ def update_status(self) -> None:
             except NotDrainedError:
                 logger.warning("Still draining shard.")
 
-    def set_password(self, username: str, password: str | None = None) -> tuple[str, str]:
-        """Handler for the set password action."""
+    def set_charmed_user_password(self, user: MongoDBUser, new_password: str) -> None:
+        """Handler for the set password."""
         self.assert_pass_password_checks()
```

Comment:
This check should be moved way higher in the call stack I think.
It used to be the safety check before settting a password in the `set-password` action. So I would call it in the config-changed / leader-elected events  maybe?